src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -4757,6 +4757,120 @@ test_qe_docs_example (void *unused)
    mongoc_client_destroy (client);
 }
 
+struct kms_callback_data {
+   int value;
+   const char *set_error;
+};
+
+static bool
+_kms_callback (void *userdata, bson_t *out, bson_error_t *error)
+{
+   struct kms_callback_data *ctx = userdata;
+   ctx->value = 42;
+   if (ctx->set_error) {
+      bson_set_error (error,
+                      MONGOC_ERROR_CLIENT_SIDE_ENCRYPTION,
+                      1729,
+                      "%s",
+                      ctx->set_error);
+      return false;
+   }
```

Comment:
```suggestion
   }
   uint8_t data[96] = {0};
   BCON_APPEND (out,
                "local",
                "{",
                "key",
                BCON_BIN (BSON_SUBTYPE_BINARY, data, sizeof (data)),
                "}");
```

Consider adding successful case with the 96 byte "key".